### ISSUE TITLE
get property by index

### ISSUE THREAD

**Issue by rkd77** [mozilla-spidermonkey/spidermonkey-embedding-examples#45]
In mozjs52 JSClass had method getProperty. Newer versions have not. Could you add an example how to properly access elements by index?
For example getElementsByTagName() returns array. a = document.getElementsByTagName("H1")[5];

**Comment by evilpie** [comment]
Do you mean something like `JS_GetElement` that takes an `uint32_t index`?

**Comment by rkd77** [comment]
I don't know. getElementsByTagName(...) returns some object. and now I want to code somehow class for this object, so it can be accessed by index. In mozjs52 I had:
```c++
static bool
htmlCollection_get_property(JSContext *ctx, JS::HandleObject hobj, JS::HandleId hid, JS::MutableHandleValue hvp)
{
   jsid id = hid.get();
...
   if (JSID_IS_INT(id)) {
                JS::RootedValue r_idval(ctx, idval);
                JS_IdToValue(ctx, id, &r_idval);
                int index = r_idval.toInt32();
                return htmlCollection_item2(ctx, hobj, index, hvp);
        }
...
}

JSClassOps htmlCollection_ops = {
  JS_PropertyStub, nullptr,
  htmlCollection_get_property, JS_StrictPropertyStub,
  ...
};
```
but AFAIK, in mozjs78 there is no getProperty in JSClassOps.
So question is how to achieve similar result with mozjs78.

**Comment by ptomato** [comment]
Right, getProperty and setProperty were removed from JSClassOps. You will either have to use lazy property resolution (example https://github.com/mozilla-spidermonkey/spidermonkey-embedding-examples/blob/esr78/examples/resolve.cpp) or pre-define the properties.
